### PR TITLE
Feature: Link a RFID to Ticket

### LINKED ISSUE #206: Implement RFID functionality
## RFID Linking
An RFID should be linked to a ticket. This should be a 1 to 1 relation.

#### Deprecated Design:

An RFID key should be linked to ticket. RFID's should use a globally defined set of boolean properties. This way permissions and consumptions can be set for every RFID.

There should be endpoints to:
* Link an RFID to a ticket
* Update RFID properties
* Get RFID properties

### PR DESCRIPTION
**Pull request by bvanderdrift**
Fixes #206 

For all interactions the user should be an Admin.

Has endpoints for:

* Getting all RFID tags (GET: /rfid)
* Linking an RFID tag to a Ticket (POST: /rfid)
* Getting a ticketId using the linked RFID (GET: /rfid/{rfid})
* Getting a RFID using a ticketId (GET: /tickets/{ticketid}/rfid
* Removing a Link using a RFID (DELETE: /rfid/{rfid})
* Removing a Link using the ticketId (DELETE: /tickets/{ticketid}/rfid)

### CODE REVIEW

File: src/main/java/ch/wisv/areafiftylan/controller/TicketRestController.java [hunk 1]
```diff
@@ -25,11 +27,13 @@
 public class TicketRestController {
     private TicketService ticketService;
     private OrderService orderService;
+    private RFIDService rfidService;
 
     @Autowired
-    public TicketRestController(TicketService ticketService, OrderService orderService) {
+    public TicketRestController(TicketService ticketService, OrderService orderService, RFIDService rfidService) {
```
Comment (skamoen) [inline src/main/java/ch/wisv/areafiftylan/controller/TicketRestController.java]: No RFID stuff in the TicketController please. You can move these to endpoints in the RFIDController such as `/rfid/tickets`

File: src/main/java/ch/wisv/areafiftylan/service/RFIDService.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+/**
+ * Created by beer on 5-5-16.
+ */
+public interface RFIDService {
+    Collection<RFIDLink> getAllRFIDLinks();
+
+    Long getTicketIdByRFID(String rfid);
+
+    String getRFIDByTicketId(Long ticketId);
```
Comment (skamoen) [inline src/main/java/ch/wisv/areafiftylan/service/RFIDService.java]: What could be a usecase for this method?

File: src/main/java/ch/wisv/areafiftylan/service/RFIDServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        return getLinkByTicketId(ticketId).getRFID();
+    }
+
+    @Override
+    public boolean isRFIDUsed(String rfid) {
+        return rfidLinkRepository.findByRfid(rfid).isPresent();
+    }
+
+    @Override
+    public boolean isTicketLinked(Long ticketId) {return rfidLinkRepository.findByTicketId(ticketId).isPresent(); }
```
Comment (skamoen) [inline src/main/java/ch/wisv/areafiftylan/service/RFIDServiceImpl.java]: Weird formatting

**Comment by bvanderdrift** [comment]
Back to @skamoen

**Comment by skamoen** [comment]
LGTM. We should reconsider this being between all the other controllers/services/repositories, but that's more of a general structure discussion then an RFID matter. 

Let's wait with merging until after the database migration release.

**Comment by skamoen** [comment]
@bvanderdrift Do you want this on master, or make future pull requests to this branch (like the consumptions and such) and merge the whole thing at once? I think the second option makes more sense as we won't be using this functionality for at least a couple releases.

**Comment by bvanderdrift** [comment]
I'd like to see this on master. Consumptions, while relevant, is a completely disjunct feature. RFID can also be useful without consumptions, such as registering for the event. I think because of this this branch can be merged to master.

@skamoen could you give this a thought and merge if you agree?

**Pull request merged**